src/routing_node.rs

```diff
@@ -95,50 +95,59 @@ impl<F, G> RoutingNode<F, G> where F : Interface + 'static,
                     //println!("Event::NewMessage({:?}, bytes)", endpoint);
 
                     let self_id = self.id.clone();
-
-                    if self.bootstraps.contains(&endpoint) {
-                        if let Ok(signed_message) = decode::<SignedMessage>(&bytes) {
-                            match signed_message.get_routing_message() {
-                                Err(_) => continue,
-                                Ok(message) => {
-                                    match message.message_type {
-                                        MessageType::PutPublicIdResponse(
-                                            ref new_public_id, ref _orig_request) => {
-                                              relocated_name = Some(new_public_id.name());
-                                              info!("Received name {:?} from network; original name was {:?}",
-                                                  relocated_name, self.id.name());
-                                              break;
+                    let opt_bootstrap_name = self.bootstraps.get(&endpoint).cloned();
+
+                    if let Some(opt_name) = opt_bootstrap_name {
```

Comment:
I understand this correct, that this ensures that we only listen to NewMessage's from registered bootstrap connections; correct?